File: account-module/src/main/java/org/example/accountmodule/account/domain/Account.java
package org.example.accountmodule.account.domain;

import lombok.Builder;
import lombok.With;
import org.example.accountmodule.contact.domain.Contact;

import java.util.List;
import java.util.UUID;

@Builder
public record Account(
        UUID id,
        String name,
        String country,
        String city,
        String street,
        String streetNumber,
        String zipCode,
        String tin,
        String webUrl,
        @With List<Contact> contacts
) {
}

File: ux/src/app/models/account/account.ts
import {ContactSummary} from '../contact/contact-summary';

export type Account = {
  id?: string,
  name: string,
  country: string,
  city: string,
  street?: string,
  streetNumber?: string,
  zipCode?: string,
  tin: string,
  webUrl?: string,
  contacts?: ContactSummary[]
}
